fred2/missioncutscenesdlg.cpp

```diff
@@ -0,0 +1,479 @@
+	The_mission.cutscenes.clear();
+	for (const auto& dialog_cutscene : m_cutscenes) {
+		The_mission.cutscenes.push_back(dialog_cutscene);
+		The_mission.cutscenes.back().formula = m_cutscenes_tree.save_tree(dialog_cutscene.formula);
+	}
+
+	theApp.record_window_data(&Mission_cutscenes_wnd_data, this);
+	CDialog::OnOK();
+}
+
+void CMissionCutscenesDlg::OnButtonNewCutscene()
+{
+	int index;
+	HTREEITEM h;
+
+	m_cutscenes.emplace_back();
+	m_sig.push_back(-1);
+
+	m_cutscenes.back().type = m_display_cutscene_types;
+	strcpy_s(m_cutscenes.back().filename, "cutscene filename");
+	h = m_cutscenes_tree.insert(m_cutscenes.back().filename);
+
+	m_cutscenes_tree.item_index = -1;
+	m_cutscenes_tree.add_operator("true", h);
+	index = m_cutscenes.back().formula = m_cutscenes_tree.item_index;
+	m_cutscenes_tree.SetItemData(h, index);
+
+	m_cutscenes_tree.SelectItem(h);
+}
+
+int CMissionCutscenesDlg::handler(int code, int node)
+{
+	int i;
+
+	switch (code) {
+	case ROOT_DELETED:
+		for (i = 0; i < (int)m_cutscenes.size(); i++)
+			if (m_cutscenes[i].formula == node)
+				break;
+
+		Assert(i < (int)m_cutscenes.size());
+		m_cutscenes.erase(m_cutscenes.begin() + i);
+		m_sig.erase(m_sig.begin() + i);
+
+		return node;
+
+	default:
+		Int3();
+	}
+
+	return -1;
+}
+
+void CMissionCutscenesDlg::OnSelchangeCutsceneTypeDrop()
+{
+	HTREEITEM h, h2;
+
+	if (cur_cutscene < 0) {
+		return;
+	}
+
+	UpdateData(TRUE);
+	UpdateData(TRUE);  // doesn't seem to update unless we do it twice..
+
+	m_cutscenes[cur_cutscene].type = m_cutscene_type;
+
+	h = m_cutscenes_tree.GetSelectedItem();
+	Assert(h);
+	while ((h2 = m_cutscenes_tree.GetParentItem(h)) != 0) {
+		h = h2;
+	}
+
+	m_cutscenes_tree.DeleteItem(h);
+	cur_cutscene = -1;
+	update_cur_cutscene();
+}
+
+void CMissionCutscenesDlg::OnChangeCutsceneName()
+{
+	HTREEITEM h, h2;
+
+	if (cur_cutscene < 0) {
+		return;
+	}
+
+	UpdateData(TRUE);
+	h = m_cutscenes_tree.GetSelectedItem();
+	if (!h){
+		return;
+	}
+
+	while ((h2 = m_cutscenes_tree.GetParentItem(h)) != 0) {
+		h = h2;
+	}
+
+	m_cutscenes_tree.SetItemText(h, m_name);
+	strcpy_s(m_cutscenes[cur_cutscene].filename, m_name);
+}
+
+void CMissionCutscenesDlg::OnCancel()
+{
+	theApp.record_window_data(&Messages_wnd_data, this);
+	CDialog::OnCancel();
+}
+
+void CMissionCutscenesDlg::OnClose()
+{
+	int z;
+
+	if (query_modified()) {
+		z = MessageBox("Do you want to keep your changes?", "Close", MB_ICONQUESTION | MB_YESNOCANCEL);
+		if (z == IDCANCEL)
+			return;
+
+		if (z == IDYES) {
+			OnOk();
+			return;
+		}
+	}
+	
+	CDialog::OnClose();
+}
+
+void CMissionCutscenesDlg::insert_handler(int old, int node)
+{
+	int i;
+
+	for (i = 0; i < (int)m_cutscenes.size(); i++) {
+		if (m_cutscenes[i].formula == old) {
+			break;
+		}
+	}
+
+	Assert(i < (int)m_cutscenes.size());
+	m_cutscenes[i].formula = node;
+	return;
+}
+
+void CMissionCutscenesDlg::move_handler(int node1, int node2, bool insert_before)
+{
+	int index1, index2, s;
+	mission_cutscene g;
```

Comment:
nitpick: since in the mission goals editor the `g` stood for `goal`, this should be changed to `c` for `cutscene`

Reply:
For the record, I am absolutely not updating this. Consider it me taking a stand regarding goober's OCD holding up fully functioning PRs over things as mundane as single-letter local variable names. Take the fully-functioning feature or don't. Your choice.

Reply:
So you'll change a single-letter variable to "fu" but not to it actual abbreviation?

Reply:
Correct. Because I'm petty.

Reply:
> Correct. Because I'm petty. And I have used the same variable naming scheme for the qtFRED version I've written. Because I'm very petty.

We are amused.  Very well, I won't make this a condition of my review.  It's more useful for the project to have your pettiness on record so that other coders know what they're dealing with.